shablona/data.py

```diff
@@ -0,0 +1,144 @@
+"""shablona.data: download and read data."""
+import os
+import sys
+import contextlib
+import os.path as op
+from hashlib import md5
+from shutil import copyfileobj
+
+if sys.version_info[0] < 3:
+    from urllib2 import urlopen, urljoin
+else:
+    from urllib.request import urlopen, urljoin
+
+# Set a user-writeable file-system location to put files:
+SHABLONA_HOME = op.join(os.path.expanduser('~'), '.shablona')
```

Comment:
[Something that follows the XDG basedir standard](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).

Basically, if it's data, it goes in `$XDG_DATA_HOME`, if it's cache, it can go in `$XDG_CACHE_HOME`. If these variables are not defined, they default to `$HOME/.local/share` and `$HOME/.cache`, respectively.

There's also the [appdirs package](https://pypi.python.org/pypi/appdirs) that will find the correct location for these sorts of things on all platforms.